### ISSUE TITLE
What does the silk-bdd-result-plugin do?

### ISSUE THREAD

**Issue by DaHu4wA** [porscheinformatik/cucumber-report-db#36]
Hello!
We want to write a "plugin" that reports the results of the latest test runs to HP ALM.

Therfore I had a look at the "silk-bdd-result-plugin". Basically this module is just a JUnit test loading a scenario test result fom the DB.
The whole module has no relation to running as a plugin or reporting to SILK?

Is there a plugin mechanism in this project to realize a report to a external tool?

**Comment by klausbayrhammer** [comment]
You are right, the silk plugin is a simple java unittest which queries the `/rest/reports` api. The idea behind the silk plugin was that you could annotate scenarios with a specific tag:
```cucumber
...
@SILK_ID_12345
Scenario: Test Silk Plugin
....
```
You could then run the `SilkSynchronizer` and specify the collection and the tag as run-arguments (when starting the test from the Silk testmanager the arguments are set by default - that's why the keys are a bit cryptic). The tests then query the `/rest/reports/` api and check whether a test with the given Tag has been run succesfully and passes or fails the unit-test depending on the result of the query.

So for your use case I would suggest you take a look at the `rest/reports` REST api. I think that api could provide all the data you need. How you integrate it into HP ALM is another story. At least for integrating the cucumber-report-db results into silk it was the easiest way to provide the `SilkSynchronizer` which was executed as a unittest by the Silk Test Manager.
